tests/readmes/test_c2d_flow.py

```diff
@@ -0,0 +1,278 @@
+
+    # Publish asset with compute service on-chain.
+    # The download (access service) is automatically created, but you can explore other options as well
+    ALGO_asset = ocean.assets.create(
+        metadata=ALGO_metadata,
+        publisher_wallet=alice_wallet,
+        encrypted_files=ALGO_encrypted_files,
+        erc721_address=ALGO_nft_token.address,
+        deployed_erc20_tokens=[ALGO_datatoken],
+    )
+
+    assert ALGO_asset.did, "create algorithm unsuccessful"
+
+    # 4. Alice allows the algorithm for C2D for that data asset
+    add_publisher_trusted_algorithm(
+        DATA_asset, ALGO_asset.did, config.metadata_cache_uri
+    )
+    DATA_asset = ocean.assets.update(DATA_asset, alice_wallet)
+
+    # 5. Bob acquires datatokens for data and algorithm
+    bob_wallet = Wallet(
+        ocean.web3,
+        os.getenv("TEST_PRIVATE_KEY2"),
+        config.block_confirmations,
+        config.transaction_timeout,
+    )
+    assert bob_wallet.address
+
+    # Alice mints DATA datatokens and ALGO datatokens to Bob.
+    # Alternatively, Bob might have bought these in a market.
+    DATA_datatoken.mint(bob_wallet.address, ocean.to_wei(5), alice_wallet)
+    ALGO_datatoken.mint(bob_wallet.address, ocean.to_wei(5), alice_wallet)
+
+    # 6. Bob starts a compute job
+
+    # Convenience variables
+    DATA_did = DATA_asset.did
+    ALGO_did = ALGO_asset.did
+
+    # Operate on updated and indexed assets
+    DATA_asset = ocean.assets.resolve(DATA_did)
+    ALGO_asset = ocean.assets.resolve(ALGO_did)
+
+    compute_service = DATA_asset.get_service("compute")
+    algo_service = ALGO_asset.get_service("access")
+
+    # Pay for dataset
+    DATA_order_tx_id = ocean.assets.pay_for_service(
+        asset=DATA_asset,
+        service=compute_service,
+        wallet=bob_wallet,
+        initialize_args={
+            "compute_environment": "unused",
+            "valid_until": int((datetime.now() + timedelta(days=1)).timestamp()),
+        },
+    )
+    assert DATA_order_tx_id, "pay for dataset unsuccessful"
+
+    # Pay for algorithm
+    ALGO_order_tx_id = ocean.assets.pay_for_service(
+        asset=ALGO_asset,
+        service=algo_service,
+        wallet=bob_wallet,
+        initialize_args={
+            "valid_until": int((datetime.now() + timedelta(days=1)).timestamp())
+        },
+    )
+    assert ALGO_order_tx_id, "pay for algorithm unsuccessful"
+
+    # Start compute job
+    DATA_compute_input = ComputeInput(DATA_did, DATA_order_tx_id, compute_service.id)
+    ALGO_compute_input = ComputeInput(ALGO_did, ALGO_order_tx_id, algo_service.id)
+    job_id = ocean.compute.start(
+        consumer_wallet=bob_wallet,
+        dataset=DATA_compute_input,
+        # TODO: Update once compute environment implemented in provider
+        compute_environment="unused",
+        algorithm=ALGO_compute_input,
+    )
+    assert job_id, "start compute unsuccessful"
+
+    # Wait until job is done
+    succeeded = False
```

Comment:
so, again: fine with me to keep it here in the test, but let's discuss the readme itself.